### PR TITLE
Fix meson

### PR DESCRIPTION
**Pull request by jpc-lip6**
Meson fixes. Solves lots of linking problems, may be interesting to see if it also solve them under MacOS.

### CODE REVIEW

File: crlcore/src/ccore/Utilities.cpp [hunk 1]
```diff
@@ -256,26 +257,25 @@ namespace CRL {
...
-      // use python to find our where we should look
-      using namespace boost::process;
-      std::error_code ec;
-      ipstream out;
-      //TODO: will need fixing for windoews
-      system("python3 -c 'from coriolis import CRL; import os; print(os.path.abspath(os.path.dirname(CRL.__file__)))'", std_out > out, ec); 
-      if (ec) {
-          cerr << "Unable to find coriolis python package. Please set CORIOLIS_TOP environment variable\n";
-          exit ( 1 );
+      std::error_code     rvalue;
```
Comment (robtaylor) [inline crlcore/src/ccore/Utilities.cpp:260]: I think leave comment in.

File: cumulus/src/designflow/pnr.py [hunk 2]
```diff
@@ -65,7 +65,8 @@ def doTask ( self ):
         else:
             print( 'PnR.doTask() run in interactive CGT mode.' )
             PnR.textMode = False
-        from .. import Etesian, Anabatic, Katana, Bora, Tramontana, Tutorial, Viewer, Unicorn
+       #from .. import Etesian, Anabatic, Katana, Bora, Tramontana, Tutorial, Viewer, Unicorn
```
Comment (robtaylor) [inline cumulus/src/designflow/pnr.py:68]: Remove commented line

File: cumulus/src/designflow/pnr.py [hunk 3]
```diff
@@ -85,9 +86,9 @@ def doTask ( self ):
             unicorn.setApplicationName  ( 'cgt')
             unicorn.registerTool        ( Etesian.GraphicEtesianEngine.grab() )
             unicorn.registerTool        ( Katana.GraphicKatanaEngine.grab() )
-            unicorn.registerTool        ( Tramontana.GraphicTramontanaEngine.grab() )
+           #unicorn.registerTool        ( Tramontana.GraphicTramontanaEngine.grab() )
             unicorn.registerTool        ( Bora.GraphicBoraEngine.grab() )
-            unicorn.registerTool        ( Tutorial.GraphicTutorialEngine.grab() )
+           #unicorn.registerTool        ( Tutorial.GraphicTutorialEngine.grab() )
```
Comment (robtaylor) [inline cumulus/src/designflow/pnr.py:91]: Is there a reason to leave these lines commented out as opposed to delete entirely?

File: documentation/build.py [hunk 4]
```diff
@@ -78,7 +78,7 @@ def __init__ ( self ):
         self.onLepka  = False
         self.onDocker = False
         hostname = socket.gethostname()
-        if hostname.startswith('lepka'): self.onLepka  = True
```
Comment (robtaylor) [inline documentation/build.py:81]: We shouldn't have anything with lip6 specific hosts in the build code.

This is ok for now, as docs still need transitioning to meson

File: hurricane/src/isobar/meson.build [hunk 5]
```diff
@@ -90,4 +90,13 @@ isobar = library(
...
 
+py.extension_module(
+  'Hurricane',
+  isobar_sources,
+  dependencies: [qt_deps, py_deps, boost],
+  link_with: [isobar, hurricane],
+  include_directories: hurricane_includes,
+  install: true,
+  subdir: 'coriolis'
+)
```
Comment (robtaylor) [inline hurricane/src/isobar/meson.build:101]: Hurricane extension is defined in the hurricane top level meson.build. Ok to move, but I didn't see a corresponding delete in this patch

**Review by robtaylor — CHANGES_REQUESTED** [review]
Looks great overall, just some minor tweaks in the comments

File: crlcore/src/ccore/Utilities.cpp (hunk 1 above)
Comment (jpc-lip6) [reply crlcore/src/ccore/Utilities.cpp:260]: OK. Will put them back. What I really should do is write a comprehensive documentation about the Coriolis initialization process especially the self location.

File: cumulus/src/designflow/pnr.py (hunk 2 above)
Comment (jpc-lip6) [reply cumulus/src/designflow/pnr.py:68]: The comment is only temporary, it's the uncommented line that will go away as soon as I add support for ``tramontana`` and ``tutorial`` (they are not deprecated tools, but new one and keep it one).

File: cumulus/src/designflow/pnr.py (hunk 3 above)
Comment (jpc-lip6) [reply cumulus/src/designflow/pnr.py:91]: Same as above. It's the uncommented which is temporary.

File: hurricane/src/isobar/meson.build (hunk 5 above)
Comment (jpc-lip6) [reply hurricane/src/isobar/meson.build:101]: Oh. I didn't see it. I did expect the module to be built in the directory holding the files (like for ``viewer`` & ``analog`` (looks more "uniform" to me). Will remove it from top level. Strangely enough, I didn't see it in the ``site-packages/cotiolis/``, maybe I was too quick.

File: documentation/build.py (hunk 4 above)
Comment (jpc-lip6) [reply documentation/build.py:81]: Not even "LIP6" code. Build for my own laptop only. In order to copy the doc onto my local ``apache`` server so I can debug quick the doc (and have a local copy).

File: hurricane/src/isobar/meson.build (hunk 5 above)
Comment (robtaylor) [reply hurricane/src/isobar/meson.build:101]: As discussed, please leave at top level

File: documentation/build.py (hunk 4 above)
Comment (robtaylor) [reply documentation/build.py:81]: I think we can do something a bit better than that ;) I'll look at a local server spin up for docs as part of the meson doc support.

File: cumulus/src/designflow/pnr.py (hunk 2 above)
Comment (robtaylor) [reply cumulus/src/designflow/pnr.py:68]: mmm, surely you can just add that when it's done?

**Review by robtaylor — APPROVED** [review]

**Pull request merged**